### PR TITLE
Seo sitemap

### PR DESCRIPTION
**Pull request by luighis**
## Summary by Sourcery

Enhance the SEO capabilities of the website by implementing a detailed sitemap for multiple sections, improving metadata for better search engine visibility, and refactoring sitemap generation for efficiency.

New Features:
- Introduce a comprehensive sitemap for various sections of the website, including blog, whitepaper, learn/security, and about pages, with appropriate change frequencies and priorities.

Enhancements:
- Refactor sitemap generation functions to remove unnecessary parameters and streamline the process by using static parameters and updated data fetching methods.

Documentation:
- Update metadata titles and descriptions for various pages to improve SEO and provide clearer information about the content.

## Release Notes

- **New Features**
  - Enhanced sitemap functionality with additional entries for various routes, improving the structure and information provided.
  - Introduced new metadata generation for project pages, improving SEO and social sharing capabilities.

- **Improvements**
  - Updated metadata titles and descriptions across several pages for better clarity and relevance.
  - Simplified function signatures across multiple sitemap functions, streamlining the code and improving performance.
  - Added `changeFrequency` property to sitemap entries for better search engine optimization.

- **Bug Fixes**
  - Ensured proper handling of project content checks and redirects in the ProjectPage function.

### CODE REVIEW

File: apps/web/app/(routes)/[lang]/blog/[category]/sitemap.ts [hunk 1]
```diff
@@ -1,24 +1,19 @@
...
 import type { MetadataRoute } from 'next'
-import type { CategoryPageProps } from './page'
 
-export default async function sitemap(
-  props: CategoryPageProps,
-): Promise<MetadataRoute.Sitemap> {
-  const {
-    params: { lang },
-  } = props
+export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/web/app/(routes)/[lang]/blog/[category]/sitemap.ts:5]: @coderabbitai yes, we shouldn't also ignore the lang routes below, even that it is a true route the path without the local (due the middleware handles the redirect) we have to still add them for multilingual support. 

cc - @luighis

File: apps/web/app/(routes)/[lang]/wallet/page.tsx [hunk 2]
```diff
@@ -43,9 +43,9 @@ export default function WalletPage() {
 }
 
 export const metadata: Metadata = {
-  title: 'Wallet | Bitlauncher',
+  title: 'Wallet Balances',
   description:
-    'Be part of the intelligent future and join the Ai/Web3 revolution now!',
+    'Your balances on the Bitcash | Bitlauncher ecosystem. Connect your Bitcash and EVM wallets',
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/web/app/(routes)/[lang]/wallet/page.tsx]: @coderabbitai good finding! Are you sure those are the variants? Good providing the components having them. We have to update all of them to use `Bitlauncher | Bitcash`. The lowercase variant might be due to logic reason.

cc - @luighis

File: apps/web/app/(routes)/[lang]/whitepaper/page.tsx [hunk 3]
```diff
@@ -10,6 +10,6 @@ export default async function WhitepaperPage() {
 }
 
 export const metadata: Metadata = {
-  title: 'Bitlauncher',
-  description: 'Whitepaper',
+  title: 'Comprehensive Whitepaper',
+  description: 'Explore the detailed whitepaper of Bitlauncher, covering all aspects of our innovative platform and technology.',
 }
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/web/app/(routes)/[lang]/whitepaper/page.tsx:15]: @coderabbitai we should track it. Create an issue based on this finding and afterwards, mark this conversation as resolved.

**Review by AndlerRL — CHANGES_REQUESTED** [review]
Very good changes! I noticed a few improvements that we can make. I added additional comments to the related feedbacks. all of them were given by coderabbitai. Next time make sure to review them (mark as resolved if a feedback doesn't apply) and make conversations to move forward.

File: apps/web/app/(routes)/[lang]/blog/[category]/sitemap.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (luighis) [reply apps/web/app/(routes)/[lang]/blog/[category]/sitemap.ts:5]: @AndlerRL 
`export default function sitemap(): MetadataRoute.Sitemap {}`
The function to generate the sitemap does not receive parameters, so it is not possible to know what language is being used in the browser

**Review by AndlerRL — APPROVED** [review]
LGTM! We need to test the SEO and sitemap once it is deployed to development env. Merging!

@AnoukRImola -- please run a web scrapping test for BL pages: About, Whitepaper, Home, Bitlauncher Project Page and Blog main page, categories and topics and check for the sitemap differences between production env for SEO improvements. Additionally, to check the opengraph for Blog articles (between topics and categories), categories and topics, we need to make sure that the CMS content is appearing as expected.

**Pull request merged**